### DISCUSSION TITLE
Is it possible to include a webgl hook in a figure environment?

### DISCUSSION THREAD

**Discussion by robjhyndman** [quarto-dev/quarto-cli#7374] [Q&A] [answered]
### Description

I'm trying to include an rgl interactive plot in a quarto book, and I want to add a figure caption. But I can't get it to work. The plot appears ok, but the figure caption and associated figure label are ignored. The code is here:
https://github.com/robjhyndman/weird/blob/main/03-multivariate.qmd#L80
The resulting output is here:
https://otexts.com/weird/03-multivariate.html#rgl51283

**Comment by mcanouil** [comment, marked as answer]
I would suggest to use the New Cross reference system of the 1.4 pre-release, see <https://quarto.org/docs/prerelease/1.4/crossref.html>.

**Reply by robjhyndman** [reply]
Thanks. That's fixed the figure caption/reference problem, but now the code fold button is centered rather than left aligned. Is there a way to control that?

**Reply by mcanouil** [reply]
Could you share a smaller and self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by cscheid** [reply]
> Thanks. That's fixed the figure caption/reference problem, but now the code fold button is centered rather than left aligned. Is there a way to control that?

That's a bug!

**Reply by cscheid** [reply]
@robjhyndman In passing, this is a neat page, thanks for sharing! Minor typo: "Multivarate" in the title of section 3.6.
